File: 1603-running-sum-of-1d-array/1603-running-sum-of-1d-array.py
class Solution:
    def runningSum(self, nums: List[int]) -> List[int]:

        # Time Complexity: O(n)
        # Space Coplexity: O(n)
        for i in range(1, len(nums)):
            nums[i] = nums[i] + nums[i-1]
            
        return nums
        

File: 1603-running-sum-of-1d-array/1603-running-sum-of-1d-array.java
class Solution {
    public int[] runningSum(int[] nums) {

        // Time Complexity: O(n)
        //  Space Coplexity: O(n)
        for (int i = 1; i <= nums.length - 1; i++) {
            nums[i] = nums[i - 1] + nums[i];
        }

        return nums;
    }
}